torchgeo/datasets/pastis.py

```diff
@@ -0,0 +1,444 @@
+    ]
+    cmap = ListedColormap(
+        [
+            (0.0, 0.0, 0.0),
+            (0.6823529411764706, 0.7803921568627451, 0.9098039215686274),
+            (1.0, 0.4980392156862745, 0.054901960784313725),
+            (1.0, 0.7333333333333333, 0.47058823529411764),
+            (0.17254901960784313, 0.6274509803921569, 0.17254901960784313),
+            (0.596078431372549, 0.8745098039215686, 0.5411764705882353),
+            (0.8392156862745098, 0.15294117647058825, 0.1568627450980392),
+            (1.0, 0.596078431372549, 0.5882352941176471),
+            (0.5803921568627451, 0.403921568627451, 0.7411764705882353),
+            (0.7725490196078432, 0.6901960784313725, 0.8352941176470589),
+            (0.5490196078431373, 0.33725490196078434, 0.29411764705882354),
+            (0.7686274509803922, 0.611764705882353, 0.5803921568627451),
+            (0.8901960784313725, 0.4666666666666667, 0.7607843137254902),
+            (0.9686274509803922, 0.7137254901960784, 0.8235294117647058),
+            (0.4980392156862745, 0.4980392156862745, 0.4980392156862745),
+            (0.7803921568627451, 0.7803921568627451, 0.7803921568627451),
+            (0.7372549019607844, 0.7411764705882353, 0.13333333333333333),
+            (0.8588235294117647, 0.8588235294117647, 0.5529411764705883),
+            (0.09019607843137255, 0.7450980392156863, 0.8117647058823529),
+            (0.6196078431372549, 0.8549019607843137, 0.8980392156862745),
+        ]
+    )
+    directory = "PASTIS-R"
+    filename = "PASTIS-R.zip"
+    url = "https://zenodo.org/record/5735646/files/PASTIS-R.zip?download=1"
+    md5 = "4887513d6c2d2b07fa935d325bd53e09"
+    prefix = {
+        "s2": os.path.join("DATA_S2", "S2_"),
+        "s1a": os.path.join("DATA_S1A", "S1A_"),
+        "s1d": os.path.join("DATA_S1D", "S1D_"),
+        "semantic": os.path.join("ANNOTATIONS", "TARGET_"),
+        "instance": os.path.join("INSTANCE_ANNOTATIONS", "INSTANCES_"),
+    }
+
+    def __init__(
+        self,
+        root: str = "data",
+        folds: Sequence[int] = (0, 1, 2, 3, 4),
+        bands: str = "s2",
+        transforms: Optional[Callable[[dict[str, Tensor]], dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new PASTIS dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            folds: a sequence of integers from 0 to 4 specifying which of the five
+                dataset folds to include
+            bands: load Sentinel-1 or Sentinel-2. One of {s1a, s1d, s2}
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+        """
+        for fold in folds:
+            assert 0 <= fold < 5
```

Comment:
Could also do something like:
```suggestion
        assert set(folds) <= set(range(6))
```
This would also raise an error for something like 2.5.